### ISSUE TITLE
[v13] css modules not carried with dynamic imports

### ISSUE THREAD

**Issue by y-nk** [vercel/next.js#44994]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:06:26 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T8112
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 7.24.3
    Relevant packages:
      next: 13.1.2
      eslint-config-next: 13.1.2
      react: 18.2.0
      react-dom: 18.2.0

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-laxqe1?file=next.config.js,components/hello-world/style.module.css,components/hello-world/index.tsx,app/%5Bslug%5D/page.tsx

### To Reproduce

1. Create a component with an import to a css modules file.

2. No matter what the context (in a `page.tsx`, etc...), load this component with `await import()`

### Describe the Bug

The styles aren't injected in the page

---

Note: if we use a classic import, of course, styles are injected.

### Expected Behavior

The styles should be injected in the page

**Comment by lucchev** [comment]
Hello,

Here is another example of reproducing this bug.
https://stackblitz.com/edit/vercel-next-js-5qed6u?file=app/PageDynamicContent.tsx

You can see the flashing of the fourth block using CSS modules.
 
The CSS code of the CSS module is not linked in the html page (<link> tag) in this case of dynamic import.
It is then injected by JavaScript.

Is there a specific configuration to solve this problem? If anyone from the Next team passes by...

**Comment by rafalwawrzyk** [comment]
Is there any update about this issue? I think it's quite urgent to fix this

**Comment by rafalwawrzyk** [comment]
Any update on this?

**Comment by y-nk** [comment]
@rafalwawrzyk in case your use-case is to load mdx with styling and everything, i've built a repo which has SSG mdx support with css modules which you can find here https://github.com/y-nk/reblog (it works by not using @next/mdx and not loading dynamically mdx with await import, but rather setting working webpack loaders which won't break the chain of dynamic imports)

**Comment by ArthurClemens** [comment]
I am seeing a similar FOUC when using a server component that imports styles from a CSS module. It happens always when the page is fetching new (uncached) data, but it can also happen randomly on other server pages. The CSS is being attached to the head only after loading is completed.

A workaround is to move these CSS module styles to global.css, but that negates the whole idea of CSS modules.

**Comment by justrealmilk** [comment]
Also subject to this.
 
- Hard navigation applies both class names to elements and includes the appropriate stylesheets.
- Client-side navigation applies class names and **doesn't inject the stylesheets**.

**Comment by RisingGeek** [comment]
> Hello,
> 
> Here is another example of reproducing a similar bug (dynamic import + css modules). https://stackblitz.com/edit/vercel-next-js-clzexc?file=app/PageDynamicContent.tsx
> 
> You can see the flashing of the fourth block using CSS modules.
> 
> The CSS code of the CSS module is not linked in the html page (link tag) in this case of dynamic import. It is injected afterwards by JavaScript.
> 
> Is there a specific configuration to solve this problem? If anyone from the Next team passes by...

@lucchev have you found any solution for this?

**Comment by joaogarin** [comment]
The same problem happens to me with pages directory and css modules in next@14.2.3

**Comment by nitroo88** [comment]
I have the same problems. First entry for a website is ok, but when I'm switching the page it will generate content without any styles. I'm working on `next start` command, because with `next dev` it works correctly.

How much long should we wait for a fix?

**Comment by joaogarin** [comment]
Could it be this one is also getting picked up here? https://github.com/vercel/next.js/pull/68396

**Comment by devjiwonchoi** [comment]
@joaogarin Can you share a repro? I couldn't reproduce this issue on v14.2.3.
x-ref: https://github.com/devjiwonchoi/repro-next-44994

**Comment by joaogarin** [comment]
I couldn't replicate it today trying in a new project. In my project I removed the problematic dynamic cals..I tried again adding them didn't see the same problems but this seemed to ahppen a bit randomly.so Not quite sure..

For now I'm good with this..looks like it might have been addressed.

**Comment by devjiwonchoi** [comment]
@joaogarin I'll do my best what I can do to resolve your issue, but without me able to reproduce, I cannot identify what's broken.. Could you try again to create a reproduction based on your existing project? 🙏

**Comment by nitroo88** [comment]
@joaogarin, everything works fine only with the experimental flag:
`experimental: {
    cssChunking: 'strict',
  }`

**Comment by marcoshernanz** [comment]
Thanks for the report. I reproduced this on 13.1.2, but the original `await import()` cases were fixed in #51018 and #52795, and the related App Router `next/dynamic` CSS case was fixed in #64294 and #64021. I also verified the minimal reproduction on the latest canary in development and production with both Turbopack and Webpack, including client navigation, and the CSS is present and applied in every case. Closing as fixed. If this still reproduces, please open a new issue with a current minimal reproduction and include the router, bundler, and mode.
